counsel.el

```diff
@@ -1695,6 +1700,15 @@ Skip some dotfiles unless `ivy-text' requires them."
 (defvar counsel-find-file-speedup-remote t
   "Speed up opening remote files by disabling `find-file-hook' for them.")
 
+(defun counsel--preselect-file ()
+  (or
+    (when counsel-find-file-at-point
+      (require 'ffap)
+      (let ((f (ffap-guesser)))
+        (when f (expand-file-name f))))
+    (when counsel-preselect-current-file
+      (when buffer-file-name (file-name-nondirectory buffer-file-name)))))
```

Comment:
This last operand of `or` can be written as

```el
(and counsel-preselect-current-file
     buffer-file-name
     (file-name-nondirectory buffer-file-name))
```

Either way, please add a docstring!

Reply:
Added docstring. I'm not convinced by `(and x y z)` being more readable but if that's more idiomatic lisp I can change it.

Reply:
My opinion may not count for much, but since we're interested in the final expression's value and not control flow, two nested `when` forms is a big no-no. Please use `and`, which is not only more idiomatic Elisp but both logically and syntactically simpler.

Reply:
"big no-no" sounds very serious. I'll change it to `and`, then.

Reply:
> "big no-no" sounds very serious

I didn't intend for it to sound so serious/harmful, sorry about that. My argument can be summarised as "it's not idiomatic and it's more complex."

Reply:
No worries :-)
I appreciate your comments and will gladly learn from you.